libnmstate/nm/connection.py

```diff
@@ -222,11 +229,19 @@ def _active_connection_callback(self, src_object, result, user_data):
             elif ac.is_activating:
                 self.waitfor_active_connection_async(ac)
             else:
-                self._mainloop.quit(
-                    'Connection activation failed on {}: reason={}'.format(
-                        ac.devname, ac.reason
+                if ac.reason == NM_DEVICE_DISCONNECTED:
+                    # When master is activating, its slaves will be
+                    # reactivated, we might get DEVICE_DISCONNECTED on slave
+                    # here. Try activate again.
+                    if self._activate_retry_count < MAX_ACTIVATE_RETRY_COUNT:
+                        self._activate_retry_count += 1
```

Comment:
On success it needs to be reset. We cannot assume that the object will not be reused for the same purpose.